manage_breast_screening/auth/oauth.py

```diff
@@ -0,0 +1,51 @@
+from authlib.integrations.django_client import OAuth
+from authlib.jose import JsonWebKey
+from authlib.oauth2.rfc7523 import PrivateKeyJWT
+from cryptography.hazmat.primitives import serialization
+from django.conf import settings
+
+oauth = OAuth()
+
+
+def get_cis2_client():
+    """Return CIS2 OAuth client configured for private_key_jwt."""
+    client = oauth.create_client("cis2")
+    metadata = getattr(client, "server_metadata", None) or {}
+    token_endpoint = metadata.get("token_endpoint")
+    if token_endpoint:
+        try:
+            headers = {}
+            jwk = jwk_from_private_key()
+            kid = jwk.thumbprint() if jwk else None
+            if kid:
+                headers["kid"] = kid
+            client.register_client_auth_method(
```

Comment:
Suggestion: I think it would be helpful here if we can link each bit of client configuration to a documentation page for CIS2, since the docs do go into a lot of detail about what the requirements are, and this might help someone who needs to review/debug this later down the line.